tsdbutil/buffer.go

```diff
@@ -2,21 +2,33 @@ package tsdbutil
 
 import (
 	"math"
-
-	"github.com/prometheus/tsdb"
 )
 
+// SeriesIterator iterates over the data of a time series.
+type SeriesIterator interface {
+	// Seek advances the iterator forward to the given timestamp.
+	// If there's no value exactly at t, it advances to the first value
+	// after t.
+	Seek(t int64) bool
+	// At returns the current timestamp/value pair.
+	At() (t int64, v float64)
+	// Next advances the iterator by one.
+	Next() bool
+	// Err returns the current error.
+	Err() error
+}
+
 // BufferedSeriesIterator wraps an iterator with a look-back buffer.
 type BufferedSeriesIterator struct {
-	it  tsdb.SeriesIterator
+	it  SeriesIterator
 	buf *sampleRing
 
 	lastTime int64
 }
 
 // NewBuffer returns a new iterator that buffers the values within the time range
 // of the current element and the duration of delta before.
-func NewBuffer(it tsdb.SeriesIterator, delta int64) *BufferedSeriesIterator {
+func NewBuffer(it SeriesIterator, delta int64) *BufferedSeriesIterator {
```

Comment:
@fabxc this is an API change so I am thinking to try the versioning with this one.

This is a not breakin change so should bump the minor version. Since we don't have any versioning shall we tag this as 2.0.1 ?

this change is needed to avoid dependancy import cycles with `tsdbutil/chunks.go`.